### PR TITLE
Torpedo Board (full pose)

### PR DESCRIPTION
**Pull request by DSsoto**
Estimate full (6DOF) pose of the torpedo board and locations of the centers of the individual targets.
I installed clang-format but still haven't figured out how to use it, some of the whitespace, and line length formatting is not ideal. I'll fix it sometime soon.
In it's next version I want to Kalman filter the 3D corners, if necessary. We'll find out at the next pool day.
ezgif com-gif-maker

### CODE REVIEW

**Comment by DSsoto** [comment]
May have inadvertently done some git foolishness because of forgetting to rebase before pushing. @jpanikulam

File: gnc/sub8_perception/include/sub8_pcl/cv_tools.hpp [hunk 1]
```diff
@@ -119,4 +126,19 @@ void range_from_param(std::string &param_root, Range &range);
...
+/// Templated function implementation
+template<typename _Matrix_Type_>
+bool pseudoInverse(const _Matrix_Type_ &a, _Matrix_Type_ &result, 
+				   double epsilon = std::numeric_limits<typename _Matrix_Type_::Scalar>::epsilon()){
+  if(a.rows()<a.cols())
+      return false;
+
+  Eigen::JacobiSVD< _Matrix_Type_ > svd = a.jacobiSvd();
+
+  typename _Matrix_Type_::Scalar tolerance =
```
Comment (jpanikulam) [inline gnc/sub8_perception/include/sub8_pcl/cv_tools.hpp:138]: wow!

File: gnc/sub8_perception/src/sub8_pcl/torpedo_board.cc [hunk 2]
```diff
@@ -1,24 +1,60 @@
...
+  log_msg << "Torpedo Board Detector subscriptions:" << std::left << std::endl
+          << std::setw(7) << "" << std::setw(8) << " left  = " << img_topic_left << std::endl 
+          << std::setw(7) << "" << std::setw(8) << " right = " << img_topic_right;
+  ROS_INFO(log_msg.str().c_str());
+
+  // Advertise Detector Service
+  service = nh.advertiseService(service_name, &Sub8TorpedoBoardDetector::request_torpedo_board_position, this);
+  log_msg.str(""); 
+  log_msg.clear();
+  log_msg << "Advertisied Service:" << std::left  << std::endl << std::setw(7) << "" << service_name;
```
Comment (jpanikulam) [inline gnc/sub8_perception/src/sub8_pcl/torpedo_board.cc:30]: typo

**Comment by jpanikulam** [comment]
Very weird git stuff.

Merging, I would suggest you read a bit more about git to avoid these things!

**Pull request merged**